### PR TITLE
[improvement](scan) speed up inserting strings into ColumnString

### PR DESCRIPTION
**Pull request by mrhhsg**
# Proposed changes
Test sql:
```sql
select count(distinct url) from hits where UserID % 5 = 0;
```
without this opt:
image

with this opt:
image

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/columns/predicate_column.h [hunk 1]
```diff
@@ -91,13 +91,17 @@ class PredicateColumnType final : public COWHelper<IColumn, PredicateColumnType<
...
         StringRef refs[sel_size];
+        size_t length = 0;
         for (size_t i = 0; i < sel_size; i++) {
             uint16_t n = sel[i];
             auto& sv = reinterpret_cast<StringValue&>(data[n]);
             refs[i].data = sv.ptr;
             refs[i].size = sv.len;
+            length += sv.len;
         }
-        res_ptr->insert_many_continuous_strings(refs, sel_size);
```
Comment (HappenLee) [inline be/src/vec/columns/predicate_column.h:100]: only here to call `insert_many_continuous_strings` after delete the code, we should del the funciton

**Review by HappenLee — COMMENTED** [review]
```
   virtual void insert_many_fix_len_data(const char* pos, size_t num) 
    virtual void insert_many_dict_data(const int32_t* data_array, size_t start_index,
                                       const StringRef* dict, size_t data_num,
                                       uint32_t dict_num = 0) {
    virtual void insert_many_binary_data(char* data_array, uint32_t* len_array,
                                         uint32_t* start_offset_array, size_t num) 
    virtual void insert_many_strings(const StringRef* strings, size_t num) 
    // Here `pos` points to the memory data type is the same as the data type of the column.
    // This function is used by `insert_keys_into_columns` in AggregationNode.
    virtual void insert_many_raw_data(const char* pos, size_t num) 
    void insert_many_data(const char* pos, size_t length, size_t data_num) 
```
Now we have lots of interface of insert ? I think we should add comments and refactor the interface to remove redundant interfaces such as `insert_elements, insert_many_strings?`

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.78 seconds
 load time: 569 seconds
 storage size: 17154699360 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221027220431_clickbench_pr_34852.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.15 seconds
 load time: 575 seconds
 storage size: 17154820913 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028091430_clickbench_pr_35001.html

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**